Expensify/App
# [No QA] Performance tracking with flipper-plugin-performance

cc @marcaaron 

### Details

Adds the Flipper side of `react-native-performance` where we can see a timeline of our performance updates
![image](https://user-images.githubusercontent.com/12156624/130151608-8cef8a18-a7ce-4207-97dc-96e3011f87d8.png)

In order to see some data on the timeline I've integrated the `<Performance>` component similar to how
it's used on the example project here: https://github.com/oblador/react-native-performance/blob/ccfe4557350a0ef206268e9a50317d9cc2480464/examples/vanilla/App.tsx#L28

### Related Issues

$ https://github.com/Expensify/App/issues/4656
$ https://github.com/Expensify/App/issues/4549
$ https://github.com/Expensify/react-native-onyx/pull/101 (package hash should be updated after the onyx PR is merged)

### Tests

1. Update local `.env` configuration and set `CAPTURE_METRICS` to `true`
2. Reset the metro cache since there's a babel config change `npm start -- --reset-cache`
3. Build a debug build with bundled JS (optional) - see how here: https://github.com/Expensify/App/issues/4656#issuecomment-901988765
4. Launch Flipper (desktop app) and install the "Performance" plugin. Search for `flipper-plugin-performance` 
5. Start the App (kill the metro server if you opted to use the bundled JS bytecode)
6. Observe data on the Performance timeline in Flipper

### QA Steps

N/A

### Tested On


### Screenshots


#### Web

![image](https://user-images.githubusercontent.com/12156624/130200630-6f027f21-0ae6-48cf-afa2-125de1d48fdc.png)

#### Mobile Web

![image](https://user-images.githubusercontent.com/12156624/130201060-fa9ffe07-678f-4dcd-aa0d-2d33c7cd1949.png)

#### Desktop

![image](https://user-images.githubusercontent.com/12156624/130197387-20957802-20b6-4ad9-b23c-02a77954f677.png)

#### iOS

![image](https://user-images.githubusercontent.com/12156624/130196817-bdd8d4e0-66ba-463e-9877-f81d5fe2b376.png)

#### Android

![image](https://user-images.githubusercontent.com/12156624/130201226-4f370dff-5d3d-48a4-aafe-bea2b7fe2ad8.png)
